cmd/flightctl-alert-exporter/main.go

```diff
@@ -39,59 +70,97 @@ func main() {
 	log.SetLevel(logLvl)
 
 	tracerShutdown := tracing.InitTracer(log, cfg, "flightctl-alert-exporter")
-	defer func() {
-		if err := tracerShutdown(ctx); err != nil {
-			log.Fatalf("failed to shut down tracer: %v", err)
-		}
-	}()
+	if tracerShutdown != nil {
+		cleanupFuncs = append(cleanupFuncs, func() error {
+			log.Info("Shutting down tracer")
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			return tracerShutdown(ctx)
+		})
+	}
 
 	ctx = context.WithValue(ctx, consts.EventSourceComponentCtxKey, "flightctl-alert-exporter")
 	ctx = context.WithValue(ctx, consts.EventActorCtxKey, "service:flightctl-alert-exporter")
 	ctx = context.WithValue(ctx, consts.InternalRequestCtxKey, true)
 
-	log.Println("Initializing data store")
+	log.Info("Initializing data store")
 	db, err := store.InitDB(cfg, log)
 	if err != nil {
-		log.Fatalf("initializing data store: %v", err)
+		return fmt.Errorf("initializing data store: %w", err)
 	}
 
 	store := store.NewStore(db, log.WithField("pkg", "store"))
-	defer store.Close()
+	cleanupFuncs = append(cleanupFuncs, func() error {
+		log.Info("Closing database connections")
```

Comment:
i would expect this to print when store.Close() returned. the caller should be extremely borning